tests/test_block_matrix.py

```diff
@@ -7,6 +7,10 @@
 
 @pytest.fixture
 def sample_matrix():
+    # YZ: This is a bit unrealistic groups distribution, which breaks my implicit
+    # assumption that diagonal blocks (e.g. bmat[3,3]) must be square matrix.
+    # This assumption should be checked in init, thus the tests should be changed.
+    # Not doing it now but we need to consider it quite soon.
```

Comment:
this is a TODO, but I will address it when cover BlockStorageMatrix with tests heavily, we will need to at some point

Reply:
Sounds good. Let's explicitly make it a todo, or even better, a GH issue.